nBTC/tests/redeem_workflow_tests.move

```diff
@@ -232,3 +289,98 @@ fun test_cannot_propose_overlapping_locked_utxos() {
 
     abort
 }
+
+#[test]
+fun test_confirm_redeem_burns_nbtc_and_removes_utxos() {
+    let (mut lc, mut ctr, redeem_id, dwallet_id, mut scenario, clock) = setup_redeem_test(
+        2500,
+        1000,
+        NBTC_P2WPKH_SCRIPT,
+        NBTC_PUBLIC_KEY,
+        true,
+    );
+
+    let r = ctr.redeem_request(redeem_id);
+    let tx_bytes = r.raw_signed_tx(ctr.storage());
+    let tx = tx::decode(tx_bytes);
+    let tx_id = tx.tx_id();
+    let parent_hash = lc.head_hash();
+    let mut header_bytes = vector[];
+    header_bytes.append(x"00000020");
+    header_bytes.append(parent_hash);
+    header_bytes.append(tx_id);
+    header_bytes.append(x"132ae858");
+    header_bytes.append(x"ffff7f20");
+    header_bytes.append(x"01000000");
+
+    let header = bitcoin_lib::header::new(header_bytes);
+    lc.insert_headers(vector[header]);
+
+    let supply_before = ctr.total_supply();
+    ctr.confirm_redeem(&lc, redeem_id, vector[], 1, 0);
+    let supply_after = ctr.total_supply();
+    assert_eq!(supply_after, supply_before - 1000);
+
+    let request = ctr.redeem_request(redeem_id);
+    assert_eq!(request.status().is_confirmed(), true);
+
+    let utxo_store = ctr.borrow_utxo_map_for_test();
+    assert_eq!(utxo_store.contains(utxo_key(0, dwallet_id)), false);
+
+    let active_dwallet = ctr.active_dwallet_id();
+    assert_eq!(utxo_store.contains(utxo_key(1, active_dwallet)), true);
+    let change_utxo = utxo_store.get_utxo(1, active_dwallet);
+    assert_eq!(change_utxo.value(), 1500);
+
+    clock.destroy_for_testing();
+    destroy(lc);
+    destroy(ctr);
+    scenario.end();
+}
+
+#[test]
+fun test_confirm_redeem_no_change() {
+    let (mut lc, mut ctr, redeem_id, dwallet_id, mut scenario, clock) = setup_redeem_test(
+        1000,
+        1000,
+        NBTC_P2WPKH_SCRIPT,
+        NBTC_PUBLIC_KEY,
+        true,
+    );
+
+    let r = ctr.redeem_request(redeem_id);
+    let tx_bytes = r.raw_signed_tx(ctr.storage());
+    let tx = tx::decode(tx_bytes);
+    let tx_id = tx.tx_id();
+
+    let parent_hash = lc.head_hash();
+    let mut header_bytes = vector[];
+    header_bytes.append(x"00000020");
```

Comment:
You can add a create function in the header.move instead of push data like this.